internal/docs/design.md

```diff
@@ -553,12 +553,15 @@ names, metadata keys, etc.). A couple of specific examples:
 
 These differences lead to a loss of portability and predictability for users.
 
-To resolve this issue, we insist that Go CDK can handle any string, and force
-drivers to use escaping mechanisms to handle strings that the underlying
+To resolve this issue, we insist that Go CDK can handle any UTF-8 string, and
```

Comment:
I think this is a reasonable choice, but it does mean that we will fail if someone is using binary. That may be very unlikely for attribute keys but less so for values.

We should at least document that here.

Actually, encoding of a non-UTF-8 string will only fail for providers that do escaping, which is not ideal. Maybe all providers should check that their input is valid UTF-8.

Reply:
I documented that the behavior for non-UTF-8 strings is currently undefined and filed an issue to consider erroring out in the concrete type.